storage/backup/backup-cli/src/coordinators/restore.rs

```diff
@@ -164,34 +217,129 @@ impl RestoreCoordinator {
             None
         };
 
-        StateSnapshotRestoreController::new(
-            StateSnapshotRestoreOpt {
-                manifest_handle: state_snapshot_backup.manifest,
-                version,
-                validate_modules: false,
-            },
-            self.global_opt.clone(),
-            Arc::clone(&self.storage),
-            epoch_history.clone(),
-        )
-        .run()
-        .await?;
+        // Restore the the state kv between lhs and rs
+        if do_phase_1 {
+            info!(
+                "Start restoring DB from version {} to tree snapshot version {}",
+                txn_start_version, tree_snapshot.version,
+            );
 
-        let txn_manifests = transaction_backups
-            .iter()
-            .map(|e| e.manifest.clone())
-            .collect();
-        TransactionRestoreBatchController::new(
-            self.global_opt,
-            self.storage,
-            txn_manifests,
-            None,
-            epoch_history,
-            VerifyExecutionMode::NoVerify,
-            None,
-        )
-        .run()
-        .await?;
+            // phase 1.a: restore the kv snapshot
+            if kv_snapshot.is_some() {
+                let kv_snapshot = kv_snapshot.clone().unwrap();
+                info!("Start restoring KV snapshot at {}", kv_snapshot.version);
+
+                StateSnapshotRestoreController::new(
+                    StateSnapshotRestoreOpt {
+                        manifest_handle: kv_snapshot.manifest,
+                        version: kv_snapshot.version,
+                        validate_modules: false,
+                        restore_mode: StateSnapshotRestoreMode::KvOnly,
+                    },
+                    self.global_opt.clone(),
+                    Arc::clone(&self.storage),
+                    epoch_history.clone(),
+                )
+                .run()
+                .await?;
+            }
+
+            // phase 1.b: save the txn between the first txn of the first chunk and the tree snapshot
+            let txn_manifests = transaction_backups
```

Comment:
add an assert that when `db_next_ver == 0 || txn_manifests.first().first_version <= db_next_ver`, just to be defensive.